### ISSUE TITLE
partial credit when 6-field lines have bad labels

### ISSUE THREAD

**Issue by yakra** [TravelMapping/DataProcessing#371]
**Consider any of the following:**
• `PA I-90 32 MA I-90 BadLabel`
• `PA I-90 BadLabel MA I-90 24`
• `PA I-90 BadLabel MA I-90 BadLabel`
Right now, since not all of the line can be read, the whole thing is thrown out and we don't get any credit for I-90.
But even if we're going from *somewhere* in Pennsylvania to *somewhere* in Massachusetts, we know that all of New York is traveled.

**Give travelers partial credit when:**
* [ ] Waypoint label... not found in line
* [ ] duplicate label

...making sure to still write a .log file error, of course.

The changes to mark_connected_route_segments.cpp will get pretty hairy pretty fast.
Don't want to break support for "reversed" list lines (e.g. MA I-90 Foo PA I-90 Bar) or splitregion.

**Comment by jteresco** [comment]
I certainly don't object to you doing this, but I also think a case can be made that a bad line, even one that spans regions,  reasonably renders the whole entry invalid.  Other than the fact that it seems a little messy to get this right, the only other reason not to do it that I have come up with is that a user with a long enough segment on a line to span three regions or more is more likely to notice the whole thing is missing than just segments near the endpoints.

**Comment by yakra** [comment]
> it seems a little messy to get this right

Yup. It was already plenty messy getting the 6-line routines ([C++](https://github.com/TravelMapping/DataProcessing/blob/5d7e7cefab6820109c0610e87a5ba284be6ba40e/siteupdate/cplusplus/classes/TravelerList/mark_connected_route_segments.cpp), [Py](https://github.com/TravelMapping/DataProcessing/blob/5d7e7cefab6820109c0610e87a5ba284be6ba40e/siteupdate/python-teresco/siteupdate.py#L1397-L1527)) right. I have to be very careful with any changes I make, especially in the C++ version.

> a case can be made that a bad line, even one that spans regions, reasonably renders the whole entry invalid. ... a user with a long enough segment on a line to span three regions or more is more likely to notice the whole thing is missing than just segments near the endpoints.

Two regions or more. :)
Though yes, you bring up a very valid point.
Perhaps it's something to get some opinions on in the forum.
